### DISCUSSION TITLE
Replace PulseAudio with Pipewire

### DISCUSSION THREAD

**Discussion by claudiux** [linuxmint/discussions#462] [Tutorials]
# Replace PulseAudio with PipeWire
[PulseAudio](https://translate.google.com/website?sl=auto&tl=fr&hl=fr&client=webapp&u=https://fr.wikipedia.org/wiki/PulseAudio) is an audio service that allows you to stream audio locally and over a network. Since 2020, it has been gradually replaced by [PipeWire](https://translate.google.com/website?sl=auto&tl=fr&hl=fr&client=webapp&u=https://fr.wikipedia.org/wiki/PipeWire) which manages the mixing of different sources and their distribution to various outputs. PipeWire also seems to produce better quality audio (IMHO).

Here's how to replace PulseAudio with PipeWire on **Linux Mint 21.3 (Cinnamon 6.0.4)**. Also successfully tested on **Linux Mint 21.2 (Cinnamon 5.8.4**).

Verify that Pipewire is not already installed. The command:
`pactl info`
should return a line containing `Server name: PulseAudio`.

If this line contains `Server name: PulseAudio (on PipeWire 0.3.48)` there is nothing more to do; Pipewire is already installed.

## Back up your system
Use TimeShift (or any other solution) to save the system state. In the event of a malfunction, it is thus possible to return to the previous state.

## Disable PulseAudio and its services
```
systemctl --user --now disable pulseaudio.service pulseaudio.socket
systemctl --user mask pulseaudio
```
The icons of applets using audio services disappear, this is normal.

## Install PipeWire and its services
```
apt install pipewire wireplumber pipewire-pulse
systemctl --user --now enable pipewire pipewire-pulse wireplumber
```

The icons for applets using audio services reappear.

## Remove Pulseaudio
Removing Pulseaudio must be done after installing Pipewire, otherwise Cinnamon may also be uninstalled!

`apt remove pulseaudio`

Restart the computer and check everything.
Once the computer restarts, the sound should work normally.

You can check that everything is started correctly with:

`systemctl --user status pipewire-pulse.socket pipewire-pulse.service`
which should return something like this:
```
● pipewire-pulse.socket - PipeWire PulseAudio
     Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.socket; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-12-01 08:16:29 CET; 1h 1min ago
   Triggers: ● pipewire-pulse.service
     Listen: /run/user/1000/pulse/native (Stream)
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/pipewire-pulse.socket

déc. 01 08:16:29 tuxc2 systemd[1912]: Listening on PipeWire PulseAudio.

● pipewire-pulse.service - PipeWire PulseAudio
     Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-12-01 08:16:29 CET; 1h 1min ago
TriggeredBy: ● pipewire-pulse.socket
   Main PID: 1922 (pipewire-pulse)
      Tasks: 2 (limit: 18332)
     Memory: 16.4M
        CPU: 1min 5.613s
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire-pulse.service
             └─1922 /usr/bin/pipewire-pulse

... systemd[1912]: Started PipeWire PulseAudio.
... pipewire-pulse[1939]: 536870912
```
The command:
`pactl info`
should return:
```
Server string: /run/user/1000/pulse/native 
Library protocol version: 35 
Server protocol version: 35 
Local: yes 
Client index: 4105 
Tile Size: 65472 
User name: <yours> 
Name 'host: <yours> 
Server Name: PulseAudio (on PipeWire 0.3.48) 
Server Version: 15.0.0 
Default Sample Specification: float32le 2ch 48000Hz 
Default Channel Plan: front-left,front-right 
Destination default: alsa_output.pci-0000_00_1b.0.analog-stereo 
Default source: alsa_input.pci-0000_00_1b.0.analog-stereo 
Cookie: ****:****
```

## Check that you don't have any packages that depend on pulseaudio

  `apt-cache rdepends --installed pulseaudio`

Note: in the list that appears, some packages depend on `pulseaudio-utils`, which does not depend on `pulseaudio`. These packages are not troublesome.

**Comment by mtwebster** [comment]
This is already done for Mint 22

**Comment by claudiux** [comment]
Okay, but LM22 is not available at the moment.

**Comment by baaannnaaannn** [comment]
> # Replace PulseAudio with PipeWire
> 
> [PulseAudio](https://translate.google.com/website?sl=auto&tl=fr&hl=fr&client=webapp&u=https://fr.wikipedia.org/wiki/PulseAudio) is an audio service that allows you to stream audio locally and over a network. Since 2020, it has been gradually replaced by [PipeWire](https://translate.google.com/website?sl=auto&tl=fr&hl=fr&client=webapp&u=https://fr.wikipedia.org/wiki/PipeWire) which manages the mixing of different sources and their distribution to various outputs. PipeWire also seems to produce better quality audio (IMHO).
> 
> Here's how to replace PulseAudio with PipeWire on Linux Mint 21.3 (Cinnamon 6.0.4). Also successfully tested on Linux Mint 21.2 (Cinnamon 5.8.4).
> 
> Verify that Pipewire is not already installed. The command: `pactl info` should return a line containing `Server name: PulseAudio`.
> 
> If this line contains `Server name: PulseAudio (on PipeWire 0.3.48)` there is nothing more to do; Pipewire is already installed.
> ## Back up your system
> 
> Use TimeShift (or any other solution) to save the system state. In the event of a malfunction, it is thus possible to return to the previous state.
> ## Disable PulseAudio and its services
> 
> ```
> systemctl --user --now disable pulseaudio.service pulseaudio.socket
> systemctl --user mask pulseaudio
> ```
> 
> The icons of applets using audio services disappear, this is normal.
> ## Install PipeWire and its services
> 
> ```
> apt install pipewire wireplumber pipewire-pulse
> systemctl --user --now enable pipewire pipewire-pulse wireplumber
> ```
> 
> The icons for applets using audio services reappear.
> ## Remove Pulseaudio
> 
> Removing Pulseaudio must be done after installing Pipewire, otherwise Cinnamon may also be uninstalled!
> 
> `apt remove pulseaudio`
> 
> Restart the computer and check everything. Once the computer restarts, the sound should work normally.
> 
> You can check that everything is started correctly with:
> 
> `systemctl --user status pipewire-pulse.socket pipewire-pulse.service` which should return something like this:
> 
> ```
> ● pipewire-pulse.socket - PipeWire PulseAudio
>      Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.socket; enabled; vendor preset: enabled)
>      Active: active (running) since Fri 2023-12-01 08:16:29 CET; 1h 1min ago
>    Triggers: ● pipewire-pulse.service
>      Listen: /run/user/1000/pulse/native (Stream)
>      CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/pipewire-pulse.socket
> 
> déc. 01 08:16:29 tuxc2 systemd[1912]: Listening on PipeWire PulseAudio.
> 
> ● pipewire-pulse.service - PipeWire PulseAudio
>      Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.service; enabled; vendor preset: enabled)
>      Active: active (running) since Fri 2023-12-01 08:16:29 CET; 1h 1min ago
> TriggeredBy: ● pipewire-pulse.socket
>    Main PID: 1922 (pipewire-pulse)
>       Tasks: 2 (limit: 18332)
>      Memory: 16.4M
>         CPU: 1min 5.613s
>      CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire-pulse.service
>              └─1922 /usr/bin/pipewire-pulse
> 
> ... systemd[1912]: Started PipeWire PulseAudio.
> ... pipewire-pulse[1939]: 536870912
> ```
> 
> The command: `pactl info` should return:
> 
> ```
> Server string: /run/user/1000/pulse/native 
> Library protocol version: 35 
> Server protocol version: 35 
> Local: yes 
> Client index: 4105 
> Tile Size: 65472 
> User name: <yours> 
> Name 'host: <yours> 
> Server Name: PulseAudio (on PipeWire 0.3.48) 
> Server Version: 15.0.0 
> Default Sample Specification: float32le 2ch 48000Hz 
> Default Channel Plan: front-left,front-right 
> Destination default: alsa_output.pci-0000_00_1b.0.analog-stereo 
> Default source: alsa_input.pci-0000_00_1b.0.analog-stereo 
> Cookie: ****:****
> ```

DUDE U HELPED ME,TY
i found a lot guides bat all of them are outdated.ty again

**Comment by hodahafez** [comment]
Thank you so much.  Worked perfectly.

**Comment by cyber-shifter** [comment]
Thank you so much for this guide! I really appreciate it, this was exactly what I needed!
